This patch blacklist text/html from the list of documents that will be
shown automatically, the patch (the muttlib.c part) has been written by
Loïc Minier <lool@dooz.org>, I've added the documentation bit.

The patch has been forwarded upstream originally by Loïc on:
http://bugs.mutt.org/3496.

The original Debian bug for this problem is http://bugs.debian.org/611410

--- a/init.h
+++ b/init.h
@@ -1211,7 +1211,9 @@
   ** ``\fCcopiousoutput\fP'' flag set for \fIevery\fP MIME attachment it doesn't have
   ** an internal viewer defined for.  If such an entry is found, mutt will
   ** use the viewer defined in that entry to convert the body part to text
-  ** form.
+  ** form. MIME attachments with 'text' types, with the only exception
+  ** of text/html, are excluded: they will be shown as they are unless auto_view
+  ** is specified.
   */
   { "include",		DT_QUAD, R_NONE, OPT_INCLUDE, M_ASKYES },
   /*
--- a/muttlib.c
+++ b/muttlib.c
@@ -658,6 +658,9 @@
   switch (m->type)
   {
     case TYPETEXT:
+      /* we don't want to display text/html */
+      if (!ascii_strcasecmp ("html", m->subtype))
+        return 1;
       /* we can display any text, overridable by auto_view */
       return 0;
       break;
